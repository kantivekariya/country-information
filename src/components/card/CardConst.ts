export const CARD_TEST_CONSTANT = {
  cardTestId: "cardTestId",
};
