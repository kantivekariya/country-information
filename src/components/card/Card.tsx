import { Link } from "react-router-dom";
import { CARD_TEST_CONSTANT } from "./CardConst";

import "./_card.scss";

interface ICardProps {
  country: {
    cca2: string;
    flags: { svg: string };
    name: {
      official: string;
    };
    population: number;
    region: string;
    capital: string;
  };
  theme: string;
}

const Card = ({ country, theme }: ICardProps) => {
  return (
    <Link
      to={`/${country.cca2}`}
      className="link"
      data-testid={CARD_TEST_CONSTANT.cardTestId}
    >
      <div className={`card-container ${theme}`}>
        <img className="card-flag" alt="flag" src={country.flags?.svg} />
        <div className="card-description">
          <h3 className="card-title">{country.name.official}</h3>
          <div className="card-text">
            <span>Population: </span>
            {country.population.toLocaleString()}
          </div>
          <div className="card-text">
            <span>Region: </span>
            {country.region}
          </div>
          <div className="card-text">
            <span>Capital: </span>
            {country.capital}
          </div>
        </div>
      </div>
    </Link>
  );
};

export default Card;
