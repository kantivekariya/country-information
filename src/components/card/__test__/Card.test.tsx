import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import Card from "../Card";
import { CARD_TEST_CONSTANT } from "../CardConst";

afterEach(cleanup);

const cardProps = {
  country: {
    cca2: "FI",
    flags: { svg: "https://flagcdn.com/fi.svg" },
    name: {
      official: "Republic of Finland",
    },
    population: 5530719,
    region: "Europe",
    capital: "Helsinki",
  },
  theme: "light",
};

describe("Card", () => {
  it("renders card", () => {
    render(<Card {...cardProps} />, {
      wrapper: MemoryRouter,
    });
    const cardElement = screen.getByTestId(CARD_TEST_CONSTANT.cardTestId);
    expect(cardElement).toBeInTheDocument();
    expect(cardElement).not.toBeChecked();
  });

  it("Click on Card", () => {
    render(<Card {...cardProps} />, {
      wrapper: MemoryRouter,
    });
    const cardElement = screen.getByTestId(CARD_TEST_CONSTANT.cardTestId);
    fireEvent.click(cardElement);
    expect(cardElement).toHaveAttribute("href", "/FI");
    expect(cardElement).toBeInTheDocument();
    expect(cardElement).not.toBeChecked();
  });

  it("should be matches snapshot", () => {
    const { baseElement } = render(<Card {...cardProps} />, {
      wrapper: MemoryRouter,
    });
    expect(baseElement).toMatchSnapshot();
  });
});
