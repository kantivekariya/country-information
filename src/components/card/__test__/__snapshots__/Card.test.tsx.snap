// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should be matches snapshot 1`] = `
<body>
  <div>
    <a
      class="link"
      data-testid="cardTestId"
      href="/FI"
    >
      <div
        class="card-container light"
      >
        <img
          alt="flag"
          class="card-flag"
          src="https://flagcdn.com/fi.svg"
        />
        <div
          class="card-description"
        >
          <h3
            class="card-title"
          >
            Republic of Finland
          </h3>
          <div
            class="card-text"
          >
            <span>
              Population: 
            </span>
            5,530,719
          </div>
          <div
            class="card-text"
          >
            <span>
              Region: 
            </span>
            Europe
          </div>
          <div
            class="card-text"
          >
            <span>
              Capital: 
            </span>
            Helsinki
          </div>
        </div>
      </div>
    </a>
  </div>
</body>
`;
